tasks/actorstate/task.go

```diff
@@ -147,10 +147,10 @@ func (t *Task) runActorStateExtraction(ctx context.Context, ts *types.TipSet, pt
 	info := ActorInfo{
 		Actor:           act,
 		Address:         addr,
-		ParentStateRoot: pts.ParentState(),
+		ParentStateRoot: ts.ParentState(),
```

Comment:
The justification lives in #273 -- I left a couple of comments there. Based on that reasoning, I believe this change is justified, but I'd like a second set of eyes before moving ahead, given the consequences of making it.

Reply:
I don't follow the logic of why this fixes the issue. The three ActorInfo fields `ParentStateRoot`, `TipSet` and `ParentTipSet` form a consistent set. `ParentStateRoot` is the state root of `TipSet` before messages are applied and `ParentTipSet` is the tipset that contained the messages that created `ParentStateRoot`.

The change you are suggesting here keeps them consistent but just moves them all forward by one epoch. Are you thinking that there is a mismatch between the epochs we use to detect the state change and the ones we use to do the actual diff? If that were wrong then I assume we would see thousands of errors where we thought we had a state change but no change was detected.

We detect state changes at https://github.com/filecoin-project/sentinel-visor/blob/baf2c251ffab3019ee7ad752ab88ac6b35ef7eb9/chain/indexer.go#L241

Here parent = pts and child = ts which seems consistent with what we are placing into ActorInfo

Reply:
> Are you thinking that there is a mismatch between the epochs we use to detect the state change and the ones we use to do the actual diff?

Yes, here's the scenario I believe is happening:

Say an Actor was created in a message from `parent.Messages()`. This means the state for this actor exists in `child.ParentState()`. A call to `StateChangedActors` using `parent.ParentState()` and `child.ParentState()` will return the new actor in the list of changes since `child.ParentState()` contains an actor not found in `parent.ParentState()`.

Now to process this Actor, we set `ActorInfo`’s `TipSet` field to `parent` and it’s `ParentTipSet` field to `ParentTipSet.Parents` and call `ProcessActors` which inturn calls to `ActorExtractor.Extract`. The extractor calls `StateReadState` passing in `parent` (`actorinfo.ts`). To read the state `LoadActor` is called on `parent`, `LoadActor` reads `parent.ParentState` and returns error not found since the actors state only exists in `child.ParentState`

Reply:
Isn't the underlying problem as stated in the original issue: the actor has been created in a message from `parent.Messages()`, therefore it has no previous state and a naive state diff can't work? You seem to be suggesting that we diff the state after it was created to the state in the next epoch.

Reply:
Ahh, I understand my confusion: the actor task diffing logic will need to accommodate the case when an actor's state exists in the current state but not the parent state. 

I have made modifications to accommodate this in my latest commit. If this seems reasonable (:+1: :-1: ) I will regenerate the vectors and include one to cover the bug this PR fixes. (looks like I can find the heights where the errors occur [here](https://github.com/filecoin-project/sentinel-infra/issues/81#issuecomment-818679005))